source/utils/Random.cpp

```diff
@@ -36,58 +30,35 @@
 unsigned long myRandomSeed;
 const unsigned long MAX = 32768;
 
-unsigned long randgen()
+static unsigned long randgen()
 {
     myRandomSeed = myRandomSeed * 1103515245 + 12345;
-    unsigned long returnVal = (unsigned int) (myRandomSeed / 65536) % MAX;
+    //TODO: What is the purpose of the cast?
```

Comment:
> returnVal used to be an unsigned int and there was a cast to avoid a warning. Then, when returnVal became unsigned long, because no warning was triggered from smaller to bigger type, it stayed unseen.

It must be that then. Casts are rarely so clever without a comment, or at least I hope. ;)
I'd propose to remove that cast and see whether it shows any problem while we're in early 0.5.0 development. WDYT guys?